cirq-core/cirq/ops/global_phase_op.py

```diff
@@ -17,28 +17,58 @@
 import numpy as np
 
 from cirq import value, protocols
-from cirq.ops import raw_types
+from cirq._compat import deprecated
+from cirq.ops import gate_operation, raw_types
 
 if TYPE_CHECKING:
     import cirq
 
 
 @value.value_equality(approximate=True)
-class GlobalPhaseOperation(raw_types.Operation):
+class GlobalPhaseOperation(gate_operation.GateOperation):
```

Comment:
Do we need the `GlobalPhaseOperation` at all after this change? Can we just deprecate it? `GateOperation(GlobalPhaseGate(1j), [])` should work fine, similar to other gates in cirq?

Reply:
We don't. Deprecating it would be nice, but that seems like a pretty big change to force on end-users. (Same for PauliStringPhasor. For BooleanHamiltonian and ArithmeticOperation I'd vote to just deprecate those after creating corresponding gates).

Reply:
Also if we ever do get around to deprecating it, we probably want to add a `cirq.global_phase(phase)`, similar to `cirq.measure`, so that users aren't required to do `GlobalPhaseGate(phase).on()`.